examples/pipelineruns/output-pipelinerun.yaml

```diff
@@ -33,7 +33,7 @@ spec:
   - name: write-new-stuff
     image: ubuntu
     command: ['bash']
-    args: ['-c', 'echo some stuff > /workspace/damnworkspace/stuff']
+    args: ['-c', 'mkdir -p /workspace/output/workspace && cp -r /workspace/damnworkspace/. /workspace/output/workspace/ && echo some stuff > /workspace/output/workspace/stuff']
```

Comment:
About the `mkdir`, perhaps it's a different issue, but I believe a Task should never need to create `/workspace/output/<resource-name>` or the resource targetPath. 

About the `cp` command, since `targetPath` is controlled by the static `Task` spec, I wonder if the copy command here could be replaced by using a different `targetPath` in the output, e.g.

```
  outputs:
    resources:
    - name: workspace
       type: git
       targetPath: /workspace/damnworkspace
```
or even some special variable:
```
  outputs:
    resources:
    - name: workspace
       type: git
       targetPath: ${inputs.resources.workspace.targetPath}
```

The controller should then take care of making sure that the output resource is also available at `/workspace/output/<resource_name>` so that it doesn't need to have a special logic to handle `targetPath` for resources that come from an output of another task.

Even if we let the copying responsibility to the user, I wonder if we should use `ln` there instead of `cp` to avoid the extra I/O.

Heh, a lot of comments when soon enough the git resource as output will go away :P - but it will be the same for the storage resource I guess.

Reply:
> About the `mkdir`, perhaps it's a different issue, but I believe a Task should never need to create `/workspace/output/<resource-name>` or the resource targetPath.

Yes! Sent that here: https://github.com/tektoncd/pipeline/pull/1156

> About the `cp` command, since `targetPath` is controlled by the static `Task` spec, I wonder if the copy command here could be replaced by using a different `targetPath` in the output, e.g.

Yes, that should be possible and would be useful in this case. I'll clean this up once #1156 goes in, since we can then drop the mkdirs.

> Even if we let the copying responsibility to the user, I wonder if we should use `ln` there instead of `cp` to avoid the extra I/O.

Yes, I'll change this to ln as well.

Reply:
Done.